src/services/fields.ts

```diff
@@ -59,12 +61,26 @@ export function extractParserAndFieldsFromDataFrame(data: DataFrame) {
   return result;
 }
 
+const getReducerId = memoize((sortBy: SortBy) => {
+  let reducerID: ReducerID | undefined = undefined;
+  if (sortBy) {
+    // Is there a way to avoid the type assertion?
+    const values: string[] = Object.values(ReducerID);
+    if (values.includes(sortBy)) {
+      reducerID = sortBy as ReducerID;
+    }
+  }
+  return reducerID;
+});
+
 export function getFilterBreakdownValueScene(
   getTitle: (df: DataFrame) => string,
   style: DrawStyle,
-  variableName: typeof VAR_FIELDS | typeof VAR_LABELS
+  variableName: typeof VAR_FIELDS | typeof VAR_LABELS,
+  sort: SortByScene
 ) {
   return (frame: DataFrame, frameIndex: number) => {
+    const reducerID = getReducerId(sort.state.sortBy);
```

Comment:
Would it be too crazy to pass the state and not the Scene? Having scenes here makes me a bit itchy.

Reply:
This actually breaks the option from being updated on sort change :'(